src/api/java/com/minecolonies/api/tileentities/TileEntityColonyBuilding.java

```diff
@@ -514,4 +522,52 @@ public AbstractContainerMenu createMenu(final int id, @NotNull final Inventory i
     {
         return new ContainerBuildingInventory(id, inv, colonyId, getBlockPos());
     }
+
+    /**
+     * Reactivate the hut of this tileEntity.
+     * Load the schematic data and set the style correctly.
+     */
+    public void reactivate()
```

Comment:
So far we've had these kinds of utilities always in the windows/messages, shouldnt we keep that to stay consistent?

Reply:
I put it here so that it has full access to the variables of the tileEntity

Reply:
Also, because it's called from 2 different messages